core/src/commonMain/kotlin/Central.kt

```diff
@@ -1,14 +1,6 @@
 package com.juul.kable
 
-import com.benasher44.uuid.Uuid
-
-public expect class Central {
-
-    public fun scanner(
-        services: List<Uuid>? = null,
-    ): Scanner
-
-    public fun peripheral(
-        advertisement: Advertisement,
-    ): Peripheral
+public interface Central {
+    public fun scanner(): Scanner
+    public fun peripheral(advertisement: Advertisement): Peripheral
 }
```

Comment:
I'm happy to see this, but I thought there was some bug that made interfaces not work. Did I misunderstand something earlier, or did something change?

Reply:
I forget exactly the issue I was running into. But some reason changes made this possible (I'm also very happy with being able to use `interface`s — still can't in some places, but the important bits now use `interface`):

I was previously storing the platform characteristic object within `Characteristic`s and then when passed around I'd have to cast to the appropriate type to get access to that `internal` property. With the `LazyCharacteristic` approach that is now used, characteristics have no references to their underlying platform characteristic objects. It is looked up lazily. So that pure Kotlin `Characteristic` can be easily used with an `interface` and no casting is necessary.

I think I was also running into issues with `Uuid` not being seen as `UUID` when on JVM. I reported https://youtrack.jetbrains.com/issue/KTIJ-405 and they renamed the ticket title with the warning ID, so now I can just suppress it!

https://github.com/JuulLabs/kable/blob/2bc92c307450ec3c3fe47bf56e18367566bd849c/core/src/androidMain/kotlin/PlatformDescriptor.kt#L6-L12